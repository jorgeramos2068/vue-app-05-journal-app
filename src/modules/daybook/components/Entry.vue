<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: entry.id } })"
  >
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">{{ month }}</span>
      <span class="mx-1 fs-5">{{ day }}, </span>
      <span class="mx-2 fw-light">{{ year }}</span>
    </div>
    <div class="entry-description">
      {{ shortText }}
    </div>
  </div>
</template>

<script>
import getCustomDate from '../helpers/getCustomDate';

export default {
  props: {
    entry: {
      type: Object,
      required: true,
    },
  },
  computed: {
    shortText() {
      return this.entry.text.length > 150
        ? `${this.entry.text.substring(0, 150)}...`
        : this.entry.text;
    },
    day() {
      const { day } = getCustomDate(this.entry.date);
      return day;
    },
    month() {
      const { month } = getCustomDate(this.entry.date);
      return month;
    },
    year() {
      const { year } = getCustomDate(this.entry.date);
      return year;
    },
  },
};
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }

  .entry-description {
    font-size: 12px;
  }
}
</style>
