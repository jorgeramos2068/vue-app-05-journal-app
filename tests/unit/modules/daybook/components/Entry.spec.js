import { shallowMount } from '@vue/test-utils';
import Entry from '@/modules/daybook/components/Entry';
import { mockJournalState } from '../../../__mocks__/mockJournalState';

describe('Entry component', () => {
  let wrapper;
  const mockRouter = {
    push: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallowMount(Entry, {
      props: { entry: mockJournalState.entries[0] },
      global: { mocks: { $router: mockRouter } },
    });
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should redirect when clicking on entry-container', () => {
    wrapper.find('.entry-container').trigger('click');
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry',
      params: { id: mockJournalState.entries[0].id },
    });
  });

  it('should test computed properties', () => {
    expect(wrapper.vm.day).toBe(23);
    expect(wrapper.vm.month).toBe('July');
    expect(wrapper.vm.year).toBe(2021);
  });
});
