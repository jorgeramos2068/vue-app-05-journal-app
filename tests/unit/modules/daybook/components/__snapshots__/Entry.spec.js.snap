// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component should match the snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">July</span><span class="mx-1 fs-5">23, </span><span class="mx-2 fw-light">2021</span></div>
  <div class="entry-description">Test text 01</div>
</div>
`;
